pkg/bootflow/bootengine/boot_process.go

```diff
@@ -0,0 +1,78 @@
+package bootengine
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/9elements/converged-security-suite/v2/pkg/bootflow/types"
+)
+
+type BootProcess struct {
+	// Do not mutate these values from the outside, the mutability
```

Comment:
you can use an accessor func here for guaranteed immutability

Reply:
Yeah, but didn't want to decide how exactly I will handle this, yet.

Reply:
ok, just making sure it's not left as a "temporary todo" :D

Reply:
Frankly, these details does not seem a priority any soon :)
There are downsides of using an accessor. For example you cannot basically serialize a structure, which does not expose fields. I'm yet not sure how exactly I will handle this.